mlflow/utils/rest_utils.py

```diff
@@ -8,56 +8,77 @@
 from mlflow.utils.logging_utils import eprint
 
 
-def get_databricks_hostname_and_auth():
+def _fail_malformed_databricks_auth(profile):
+    raise Exception("Got malformed Databricks CLI profile '%s'. Please make sure the Databricks "
+                    "CLI is properly configured as described at "
+                    "https://github.com/databricks/databricks-cli." % profile)
+
+
+def get_databricks_http_request_params_or_fail(profile=None):
```

Comment:
Refactored method to return a blackbox object which can be passed into `requests` directly, to avoid intermediates having to know about things like `auth`, `hostname`, and `headers`.